src/Disks/createVolume.cpp

```diff
@@ -1,17 +1,57 @@
 #include "createVolume.h"
 
+#include <Disks/SingleDiskVolume.h>
+#include <Disks/VolumeJBOD.h>
+#include <Disks/VolumeRAID1.h>
+
+#include <boost/algorithm/string.hpp>
+
 namespace DB
 {
 
+namespace ErrorCodes
+{
+    extern const int UNKNOWN_RAID_TYPE;
+}
+
 VolumePtr createVolumeFromReservation(const ReservationPtr & reservation, VolumePtr other_volume)
 {
     if (other_volume->getType() == VolumeType::JBOD || other_volume->getType() == VolumeType::SINGLE_DISK)
     {
-        /// Since reservation on JBOD chosies one of disks and makes reservation there, volume
+        /// Since reservation on JBOD chooses one of disks and makes reservation there, volume
         /// for such type of reservation will be with one disk.
         return std::make_shared<SingleDiskVolume>(other_volume->getName(), reservation->getDisk());
     }
+    if (other_volume->getType() == VolumeType::RAID1)
+    {
+        auto volume = std::dynamic_pointer_cast<VolumeRAID1>(other_volume);
```

Comment:
Ugly, I think max_data_part_size can be common property in IVolume interface. Or maybe virtual method `getMaxSize` or something like this.

Reply:
At some point there would appear other volume-specific fields and it will be neccessary to cast incoming pointer like that :man_shrugging: